pkg/registry/core/pod/storage/storage.go

```diff
@@ -346,13 +346,26 @@ func (r *EphemeralContainersREST) Update(ctx context.Context, name string, objIn
 		return newPod, nil
 	})
 
-	obj, _, err = r.store.Update(ctx, name, updatedPodInfo, createValidation, updateValidation, false, options)
+	// Validation should be passed the API kind (EphemeralContainers) rather than the storage kind.
+	obj, _, err = r.store.Update(ctx, name, updatedPodInfo, toEphemeralContainersCreateValidation(createValidation), toEphemeralContainersUpdateValidation(updateValidation), false, options)
 	if err != nil {
 		return nil, false, err
 	}
 	return ephemeralContainersInPod(obj.(*api.Pod)), false, err
 }
 
+func toEphemeralContainersCreateValidation(f rest.ValidateObjectFunc) rest.ValidateObjectFunc {
+	return func(ctx context.Context, obj runtime.Object) error {
+		return f(ctx, ephemeralContainersInPod(obj.(*api.Pod)))
+	}
+}
+
+func toEphemeralContainersUpdateValidation(f rest.ValidateObjectUpdateFunc) rest.ValidateObjectUpdateFunc {
+	return func(ctx context.Context, obj, old runtime.Object) error {
+		return f(ctx, ephemeralContainersInPod(obj.(*api.Pod)), ephemeralContainersInPod(old.(*api.Pod)))
+	}
+}
+
 // Extract the list of Ephemeral Containers from a Pod
 func ephemeralContainersInPod(pod *api.Pod) *api.EphemeralContainers {
 	ephemeralContainers := pod.Spec.EphemeralContainers
```

Comment:
is there a reason the entire pod metadata isn't being copied, like this?
```
ObjectMeta: pod.ObjectMeta,
```

Reply:
I copied this behavior from scale:
 https://github.com/kubernetes/kubernetes/blob/f99ed92ed149e8cbb09dfcdcff78e6fb4024d3a0/pkg/registry/apps/deployment/storage/storage.go#L334 where there's a comment `TODO: Create a variant of ObjectMeta type that only contains the fields below.`

I'm not sure of the original reason, but I'm happy to experiment here.

Reply:
hmm... ok. the labels and annotations seem most likely to be of interest. managedFields would not be coherent if copied from the pod metadata.

Reply:
We could copy the entire ObjectMeta and prune out fields that don't make sense. I guess it's not uncommon to derive `ObjectMeta`, so maybe there should be a helper for it.

Reply:
@liggitt shall we submit this as-is to fix the bug and open a follow-up issue?

Reply:
yes, follow-up for considering what we want to do with this bit is fine.